### PR TITLE
Add chain argument to picobox.push(...)

### PR DESCRIPTION
**Pull request by ikalnytskyi**
Picobox stacked interface has one significant drawback so far: it
doesn't allow to override dependencies partially. That means you can
change a box (i.e. DI container) at the top but a new box is required
to provide the very same dependencies.

While this is enough to do basic management it's a little bit
inconvenient in some edge cases like tests. Sometime you need to only
partially override dependencies and there were no way to do so.

So this patch is about introducing 'chain' argument in 'picobox.push()'
interface which basically means "when a dependency is missed, please
lookup for it down the stack".

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -195,6 +195,29 @@ general methods that will be applied to latest active box instance.
 
     spam()                          # RuntimeError: no boxes on the stack
 
+When only partial overriding is necessary, you can chain pushed box so any
+missed look ups will be proxied to the boxes down the stack.
```
Comment (malor) [inline docs/index.rst]: nit: noun is "lookup", therefore, "lookups"

File: src/picobox/_stack.py [hunk 2]
```diff
@@ -55,19 +55,26 @@ def do(magic):
             with picobox.push(barbox):
                 assert do() == 14
             assert do() == 43
+
+    :param box: A :class:`Box` instance to push to the top of the stack.
+    :param chain: (optional) Look up missed keys down the stack if ``True``.
```
Comment (malor) [inline src/picobox/_stack.py]: nit: maybe you should explicitly mention that the keys are looked up only one level up the stack. So to look up through multiple levels, each level must be created with `chain=True`
